### PR TITLE
Nuke Consul.

### PR DESCRIPTION
**Pull request by rabbah**
A quick attempt to strip consul - mostly because I don't want to understand why it doesn't deploy with docker-for-mac.

Need to adjust all the deployment parameters (maybe cherry picking from https://github.com/apache/incubator-openwhisk/pull/2277 and merging PRs is the way to go.

Related to #2254..

@ddragosd FYI.

### CODE REVIEW

**Comment by rabbah** [comment]
one issue found is that our configuration of the db run on "localhost" but the deployment of the containers is on the remote host which means the precomputed `db_prefix` will not match.

**Comment by rabbah** [comment]
@ddragosd I didn't squash my last two commits in case they're helpful to you. I'm happy to abandon this pr as noted I'm trying to get on docker for mac and didn't want to deal with consul.

File: core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala [hunk 1]
```diff
@@ -142,7 +139,6 @@ class InvokerPool(
 
     def publishStatus() = {
         val json = invokerStatus.toMap.mapValues(_.asString).toJson
```
Comment (markusthoemmes) [inline core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala]: remove this line as well.

File: core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala [hunk 2]
```diff
@@ -142,7 +139,6 @@ class InvokerPool(
 
     def publishStatus() = {
```
Comment (markusthoemmes) [inline core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala]: Rename to `printStatus`?

**Comment by ddragosd** [comment]
You're making it easier with this PR @rabbah 😄  ; we can merge this one instead, and drop #2277. I'll try it as soon as I get a chance.

File: ansible/roles/controller/tasks/deploy.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@
     mode: 0777
   become: true
 
-- name: (re)start controller
+- name: (re)start controller "{{ db.whisk.actions }}"
```
Comment (markusthoemmes) [inline ansible/roles/controller/tasks/deploy.yml]: I assume the db print is a "debug" statement?

File: ansible/roles/invoker/tasks/deploy.yml [hunk 4]
```diff
@@ -63,16 +63,38 @@
...
         -e SELF_DOCKER_ENDPOINT=localhost
-        -e SERVICE_CHECK_HTTP=/ping
-        -e SERVICE_CHECK_TIMEOUT=2s
-        -e SERVICE_CHECK_INTERVAL=15s
-        -e JAVA_OPTS=-Xmx{{ invoker.heap }}
-        -e INVOKER_OPTS='{{ invoker.arguments }}'
+        -e DOCKER_REGISTRY={{ docker_registry }}
+        -e DOCKER_IMAGE_PREFIX={{ docker_image_prefix }}
+        -e DOCKER_IMAGE_TAG={{ docker_image_tag }}
+        -e INVOKER_CONTAINER_NETWORK=bridge
```
Comment (markusthoemmes) [inline ansible/roles/invoker/tasks/deploy.yml]: I don't think this was hardcoded to "bridge" before, was it? This needs to be configurable.

File: core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala [hunk 5]
```diff
@@ -92,14 +92,13 @@ trait ContainerUtils {
         val cpuArg = Array("-c", cpuShare.toString)
         val memoryArg = Array("-m", s"${limits.memory.megabytes}m", "--memory-swap", s"${limits.memory.megabytes}m")
         val capabilityArg = Array("--cap-drop", "NET_RAW", "--cap-drop", "NET_ADMIN")
-        val consulServiceIgnore = Array("-e", "SERVICE_IGNORE=true")
```
Comment (markusthoemmes) [inline core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala]: Needs to be changed in the new pool as well, see `DockerContainer.scala` line 75 (you probably didn't get it as a match in your directory-wide "consul" search? 😆 )

File: docs/about.md [hunk 6]
```diff
@@ -81,13 +81,9 @@ The record of the action contains mainly the code to execute (shown above) and d
 
 In this particular case, our action doesn’t take any parameters (the function’s parameter definition is an empty list), thus we assume we haven’t set any default parameters and haven’t sent any specific parameters to the action, making for the most trivial case from this point-of-view.
 
-### Who’s there to invoke the action: Consul
+### Who’s there to invoke the action: Load balancer
```
Comment (markusthoemmes) [inline docs/about.md]: Make consistent use of capitalization: Load balancer vs Load Balancer

File: docs/reference.md [hunk 7]
```diff
@@ -212,7 +212,6 @@ The following packages are available to be used in the Node.js 6.9.1 environment
 - [cheerio v0.22.0](https://www.npmjs.com/package/cheerio) - Fast, flexible & lean implementation of core jQuery designed specifically for the server.
 - [cloudant v1.6.2](https://www.npmjs.com/package/cloudant) - This is the official Cloudant library for Node.js.
 - [commander v2.9.0](https://www.npmjs.com/package/commander) - The complete solution for node.js command-line interfaces.
-- [consul v0.27.0](https://www.npmjs.com/package/consul) - A client for Consul, involving service discovery and configuration.
```
Comment (markusthoemmes) [inline docs/reference.md]: Why touch this? It's a package installed in our Node.js runtime, which has nothing to do with our system's consul.

File: ansible/roles/controller/tasks/deploy.yml (hunk 3 above)
Comment (rabbah) [inline ansible/roles/controller/tasks/deploy.yml]: yes - will remove; will need to resolve the db prefix without consul; we were getting away with this undetected.

File: core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala (hunk 5 above)
Comment (rabbah) [inline core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala]: :) thanks for the catch!!

File: core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala (hunk 2 above)
Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala]: how about `logStatus`?

File: ansible/environments/docker-machine/group_vars/all [hunk 8]
```diff
@@ -9,7 +9,7 @@ docker_dns: ""
 # a hostname that is resolved on the client, via /etc/hosts for example.
 whisk_api_localhost_name: "openwhisk"
 
-db_prefix: docker_whisk_
+db_prefix: "{{ ansible_user_id|lower }}_{{ ansible_hostname|lower }}_"
```
Comment (rabbah) [inline ansible/environments/docker-machine/group_vars/all]: this is a change we might want to keep as hard coded.

File: core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala [hunk 9]
```diff
@@ -442,16 +441,12 @@ object Invoker {
...
+        invokerUseReactivePool -> false.toString) ++
         ExecManifest.requiredProperties ++
-        WhiskAuthStore.requiredProperties ++
         WhiskEntityStore.requiredProperties ++
         WhiskActivationStore.requiredProperties ++
         ContainerPool.requiredProperties ++
-        kafkaHost ++
-        consulServer ++
-        whiskVersion
+        kafkaHost
```
Comment (ddragosd) [inline core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala]: are we leaving `whiskVersion` out ?

Comment (rabbah) [inline core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala]: it isn't used - so it's deadcode.
could add it back.

**Comment by DanLavine** [comment]
Hey all. I was just wondering what has happened to this PR? It seems like it would be greatly appreciated by everyone who does not want to heavily rely on Ansible to deploy OpenWhisk. After a couple weeks of sitting here, did it fall off of the radar?

**Comment by markusthoemmes** [comment]
@DanLavine we need to prepare some things behind the covers before we can get this in. It's well on our radar.

**Comment by rabbah** [comment]
We need to deal with this comment for docker-machine https://github.com/apache/incubator-openwhisk/pull/2452#issuecomment-312442428.

**Comment by rabbah** [comment]
Last fix broke Travis but I didn't investigate yet.

**Comment by rabbah** [comment]
an invalid value, which appears to include a variable that is undefined. The error was: 'docker_image_tag' is undefined\n\nThe error appears to have been in ...

**Comment by markusthoemmes** [comment]
@rabbah the last travis was green already, I was aware of the issue.

**Comment by rabbah** [comment]
Interesting I only got one slack notification. The second must have gone to you. I should modify the bot. Thx for handling.

**Comment by rabbah** [comment]
pg3/896 🔵

**Comment by rabbah** [comment]
Rebased to resolve last round of conflicts - also address docker-machine issue in last unsquashed commit for review.

**Comment by rabbah** [comment]
pg3/934 🚀

**Pull request merged**